src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java

```diff
@@ -1029,4 +1028,52 @@ public Collection<String> getPotentialWorlds() {
                 .map(File::getName)
                 .collect(Collectors.toList());
     }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addOrRemoveWorldSafely(String worldName, String operationName, Runnable worldModification) {
+        if (safeToAddOrRemoveWorld()) {
+            // Operation is fine to do now
+            worldModification.run();
+        } else {
+            // Operation needs to be delayed until worlds are not being ticked
+
+            Logging.fine("Worlds were being ticked while attempting to %s %s. Trying again in the next tick", operationName, worldName);
+            new BukkitRunnable() {
+                public void run() {
+                    worldModification.run();
+                }
+            }.runTask(plugin);
```

Comment:
Yes, it will always be safe to run on the next tick. The case where the modification is run immediately is meant for when players run the commands or if the command block is on a Spigot server. In those cases, it is considered safe to run the modification immediately, so we do so.

It actually isn't necessary to delay the modification at all anymore, as Paper accepted Spigot's fix, so there aren't any errors anymore. However, Paper has asked that plugins use their API to check when the worlds are being ticked and deffer when possible: https://github.com/PaperMC/Paper/pull/8316#issuecomment-1256864284 (I've updated the title to reflect this). So, this PR doesn't fix any errors anymore, but it dose comply with the Paper dev's request.

Reply:
So, the case where `worldModification` gets run immediately is so that Players can get immediate feedback. The case where `worldModification` is delayed is when the world map is being iterated through when on a Paper server, which Paper has asked to happen.